File: infrastructures/ddd/python/src/infrastructure/persistence/read/infra/base_read_entity.py
"""
Base class for all read entities (query side of CQRS).
Read entities are pre-rendered views optimized for front-end.
They are denormalized for query performance.
"""

from infrastructure.persistence.infra import BaseEntity


class BaseReadEntity(BaseEntity):
    """
    Base class for read entities (query side).
    
    Read entities:
    - Live in /infrastructure/persistence/read/
    - Have ReadEntity suffix
    - Pre-rendered views optimized for queries
    - Denormalized for performance
    - MAY aggregate data from multiple write entities
    - Returned directly to service layer (no BMO mapping in read path)
    """
    
    def __init__(self):
        """Initialize a new read entity."""
        super().__init__()

File: infrastructures/ddd/typescript/src/infrastructure/persistence/read/infra/BaseReadEntity.ts
import { BaseEntity } from '../../infra/BaseEntity';

/**
 * Base class for all Read Entities (query side of CQRS).
 * This class:
 * - Lives in /infrastructure/persistence/read/infra/
 * - Used for query operations (read-only data retrieval)
 * - Denormalized for query performance
 * - CAN have database/ORM attributes
 * - MAY aggregate from multiple WriteEntities
 * 
 * ReadEntities:
 * - Suffix with "ReadEntity"
 * - Live in /infrastructure/persistence/read/
 * - Pre-rendered views optimized for front-end
 * - Returned directly to service (no BMO in read path for performance)
 */
export abstract class BaseReadEntity extends BaseEntity {
  constructor() {
    super();
  }
}
